zeapp/android/src/main/java/de/berlindroid/zeapp/zevm/ZeBadgeViewModel.kt

```diff
@@ -363,78 +356,24 @@ class ZeBadgeViewModel
-                        slots = it.slots.copy(slot to initialConfiguration(slot)),
+                        slots = it.slots.copy(slot to defaultSlotConfiguration),
                     )
                 }
             }
         }
 
         private suspend fun initialConfiguration(slot: ZeSlot): ZeConfiguration {
-            if (preferencesService.isSlotConfigured(slot)) {
-                val configuration = preferencesService.getSlotConfiguration(slot)
-                if (configuration != null) {
-                    return configuration
-                }
-            }
-
-            return when (slot) {
-                is ZeSlot.Name ->
-                    ZeConfiguration.Name(
-                        null,
-                        null,
-                        imageProviderService.getInitialNameBitmap(),
-                    )
-
-                is ZeSlot.FirstSponsor -> ZeConfiguration.Picture(R.drawable.page_google.toBitmap())
-                is ZeSlot.FirstCustom -> ZeConfiguration.Picture(R.drawable.soon.toBitmap())
-                is ZeSlot.SecondCustom -> ZeConfiguration.Picture(R.drawable.soon.toBitmap())
-                ZeSlot.QRCode ->
-                    ZeConfiguration.QRCode(
-                        title = "",
-                        text = "",
-                        url = "",
-                        isVcard = false,
-                        phone = "",
-                        email = "",
-                        bitmap = R.drawable.qrpage_preview.toBitmap(),
-                    )
-
-                ZeSlot.Weather ->
-                    ZeConfiguration.Weather(
-                        LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE),
-                        "22C",
-                        R.drawable.soon.toBitmap(),
-                    )
-
-                is ZeSlot.Quote ->
-                    ZeConfiguration.Quote(
-                        "Test",
-                        "Author",
-                        R.drawable.page_quote_sample.toBitmap(),
-                    )
-
-                ZeSlot.BarCode ->
-                    ZeConfiguration.BarCode(
-                        "Your title for barcode",
-                        "",
-                        R.drawable.soon.toBitmap(),
-                    )
-
-                ZeSlot.Add ->
-                    ZeConfiguration.Name(
-                        null,
-                        null,
-                        imageProviderService.provideImageBitmap(R.drawable.add),
-                    )
-
-                ZeSlot.Camera ->
-                    ZeConfiguration.Camera(
-                        imageProviderService.provideImageBitmap(R.drawable.soon),
-                    )
-            }
+            // try to get from the persistence, if not present, return the default
+            return zeSlotRepository.getSlotConfiguration(slot = slot)
+                ?: zeSlotRepository.getDefaultSlotConfiguration(slot = slot)
```

Comment:
Now the initial configuration is a bit easier to understand.
We basically check if the configuration is present, if not, default to default.